Added auto detection to .mzXML files

Mailaender opened the pull request:
This is a followup of https://github.com/eclipse/chemclipse/pull/534 which brings it in line with the other recent XML file formats

Mailaender committed:
Auto detect .mzXML files.

eselmeister commented on chemclipse/plugins/org.eclipse.chemclipse.msd.converter.supplier.mzxml/src/org/eclipse/chemclipse/msd/converter/supplier/mzxml/converter/MassSpectrumMagicNumberMatcher.java:
```diff
@@ -0,0 +1,60 @@
+			JAXBContext jaxbContext = JAXBContext.newInstance(IFormat.CONTEXT_PATH_V_320);
+			Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+			MsRun msRun = (MsRun)unmarshaller.unmarshal(nodeList.item(0));
+			if(msRun.getScan().size() == 1)
```
Please put the if condition in curly brackets {}.

eselmeister commented on chemclipse/plugins/org.eclipse.chemclipse.msd.converter.supplier.mzxml/src/org/eclipse/chemclipse/msd/converter/supplier/mzxml/converter/ChromatogramMagicNumberMatcher.java:
```diff
@@ -0,0 +1,60 @@
+			JAXBContext jaxbContext = JAXBContext.newInstance(IFormat.CONTEXT_PATH_V_320);
+			Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+			MsRun msRun = (MsRun)unmarshaller.unmarshal(nodeList.item(0));
+			if(msRun.getScan().size() > 1)
```
Please put the if condition in curly brackets {}.

eselmeister commented:
Please have a look at the comment. Principally, it looks good, but having a deep look at the *.xml content could lead to performance issues. But I don't see any other way to precisely check the file.

Mailaender committed:
Code style: prefer curly braces even for one liners.

Mailaender commented:
Yes, I also felt bad about it, so I added https://github.com/eclipse/chemclipse/pull/655 to investigate. Rebased and the cost is 4 ms per file on local SSD:
> INFO  main converter.AbstractSupplierFileIdentifier.isMatchMagicNumber (AbstractSupplierFileIdentifier.java:173) Magic number check of mabr_eqa20qty_0006_2L1.mzXML by mzXML Chromatogram (*.mzXML) took 0.002 seconds.
INFO  main converter.AbstractSupplierFileIdentifier.isMatchMagicNumber (AbstractSupplierFileIdentifier.java:173) Magic number check of mabr_eqa20qty_0006_2L1.mzXML by mzXML Mass Spectra (*.mzXML) took 0.002 seconds.

which is noticeable when loading folders with >100 files e.g. https://osf.io/d6hzx/

Mailaender committed:
Fix a null pointer exception.
